oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/writer/DefaultIndexWriter.java

```diff
@@ -160,7 +160,7 @@ public boolean close(long timestamp) throws IOException {
 
     //~----------------------------------------< internal >
 
-    IndexWriter getWriter() throws IOException {
+    synchronized IndexWriter getWriter() throws IOException {
```

Comment:
I have to admit I'm not 100% confident that this change has no unwanted side effects... For example, a deadlock. Or a performance degradation, if other methods of this class are synchronized. Something to be investigated, I think... Unless, you have done that already?

Reply:
thanks Thomas, I have refined this per suggestion